File: Autolib/Station.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace Autolib
{
    public partial class Station
    {
        public Station()
        {
            Bornes = new HashSet<Borne>();
        }

        public int IdStation { get; set; }
        public decimal Latitude { get; set; }
        public decimal Longitude { get; set; }
        public string Adresse { get; set; }
        public int? Numero { get; set; }
        public string Ville { get; set; }
        public int? CodePostal { get; set; }

        public virtual ICollection<Borne> Bornes { get; set; }
    }
}

File: Autolib/ClientApp/src/app/models/station.ts
import { Borne } from "./borne";

export class Station {
  idStation: number;
  latitude: number;
  longitude: number;
  adresse: string;
  numero: number;
  ville: string;
  codePostal: number;
  bornes: Borne[];

  public totalBornes(): number {
    return this.bornes.length;
  }

  public totalBornesDispo(): number {
    let dispo = 0;
    for (let borne of this.bornes) {
      if (borne.etatBorne == 1) dispo++;
    }
    return dispo;
  }

  public getBornesDispo(): Borne[] {
    let bornesDispo: Borne[];
    for (let borne of this.bornes) {
      if (borne.etatBorne == 1) bornesDispo.push(borne);
    }
    return bornesDispo;
  }

  public getAdresse(): string {
    return this.numero.toString() + " " +
           this.adresse + ", " +
           this.codePostal.toString() + " " + this.ville; 
  }
}

